src/thor/route_action.cc

```diff
@@ -231,39 +231,50 @@ void thor_worker_t::route(Api& request) {
 
 thor::PathAlgorithm* thor_worker_t::get_path_algorithm(const std::string& routetype,
                                                        const valhalla::Location& origin,
-                                                       const valhalla::Location& destination) {
+                                                       const valhalla::Location& destination,
+                                                       const Options& options) {
+  // make sure they are all cancelable
+  for (auto* alg : std::vector<PathAlgorithm*>{
+           &multi_modal_astar,
+           &timedep_forward,
+           &timedep_reverse,
+           &astar,
+           &bidir_astar,
+           &bss_astar,
+       }) {
+    alg->set_interrupt(interrupt);
+  }
```

Comment:
this seemed preferable to setting them all individually but honestly i dont like it that much either